libraries/AP_Networking/AP_Networking_port.cpp

```diff
@@ -342,22 +347,33 @@ bool AP_Networking::Port::send_receive(void)
 
     if (connected) {
         // handle outgoing packets
-        uint32_t available = writebuffer->available();
-        available = MIN(300U, available);
+        uint32_t available;
+
+        {
+            WITH_SEMAPHORE(sem);
+            available = writebuffer->available();
+            available = MIN(300U, available);
 #if HAL_GCS_ENABLED
-        if (packetise) {
-            available = mavlink_packetise(*writebuffer, available);
-        }
+            if (packetise) {
+                available = mavlink_packetise(*writebuffer, available);
```

Comment:
Do we really need to hold the semaphore while doing the packetise?

Reply:
no, but losing/regaining it will likely cost more